### PR TITLE
Test Atlas: Apply ISO Country Codes to Features

### PR DESCRIPTION
**Pull request by Bentleysb**
### Description:

Currently there is no easy way to apply an ISO country code to all features when creating a test atlas. When using annotations it is possible to add ISO codes by manually adding them to each feature. When importing from a resource the recourse file itself must be edited to add the tag for the codes. This becomes quite cumbersome when working with large test cases that require every feature to have an ISO country code. 
This fixes this problem by tweaking the behavior of the `iso` property of the `TestAtlas` annotation. Previously this property just updated the atlas meta data to include the ISO code. Now this code is applied as a tag to all features in the test atlas, whether they are from an annotation or a resource (excluding text atlases).

### Potential Impact:

The largest potential impacts are mitigated by maintaining the do nothing if `unknown` logic. This way the ISO tags are only added when the `iso` parameter is added to a `TestAtlas` annotation. 
This is still a potentially breaking change if there is something using the `iso` parameter and is expecting a lack of ISO tags or a certain tag count/set of tags.   
It is also possible that the addition of country slicing when iso code are passed could have an impact. However no real slicing is done becasue the country boundary map that is used contains `Rectangle.MAXIMUM` as the only 'country'.

### Unit Test Approach:

Added unit tests to `AtlasTestCase` to check that this works for annotation test atlases.
Added unit tests to `TestAtlasTest` to check that this works for resources imported as test atlases. 

### Test Results:

None

------

In doubt: [Contributing Guidelines](CONTRIBUTING.md)

### CODE REVIEW

File: src/main/java/org/openstreetmap/atlas/utilities/testing/TestAtlasHandler.java [hunk 1]
```diff
@@ -91,25 +103,33 @@ public static Atlas getAtlasFromJsomOsmResource(final boolean josmFormat,
...
+                    .fromBoundaryMap(Collections.singletonMap(iso, MultiPolygon.MAXIMUM)));
+        }
         final Atlas rawAtlas = new RawAtlasGenerator(pbfResource, loadingOption,
                 MultiPolygon.MAXIMUM).build();
 
-        // Way-section
-        return new WaySectionProcessor(rawAtlas, loadingOption).run();
+        // Country Slice and Way-Section
+        return new WaySectionProcessor(iso.equals(ISOCountryTag.UNKNOWN_ISO_COUNTRY) ? rawAtlas
+                : new RawAtlasCountrySlicer(loadingOption).slice(rawAtlas), loadingOption).run();
```
Comment (nachtm) [inline src/main/java/org/openstreetmap/atlas/utilities/testing/TestAtlasHandler.java]: Is the fact that we're now country-slicing something worth calling out? Feels like that might have some downstream impacts on JOSM-atlas tests

File: src/main/java/org/openstreetmap/atlas/utilities/testing/TestAtlasHandler.java [hunk 2]
```diff
@@ -91,25 +103,33 @@ public static Atlas getAtlasFromJsomOsmResource(final boolean josmFormat,
...
     }
 
-    private static Map<String, String> mergeTags(final String[] firstTags,
-            final String[] secondTags)
+    private static String[] mergeTags(final String[] firstTags, final String[] secondTags)
     {
         final List<String> allTags = new ArrayList<>(Arrays.asList(firstTags));
         allTags.addAll(Arrays.asList(secondTags));
-        return parseTags(allTags.toArray(new String[0]));
+        return allTags.toArray(new String[0]);
```
Comment (nachtm) [inline src/main/java/org/openstreetmap/atlas/utilities/testing/TestAtlasHandler.java:134]: Why the change here?

File: src/main/java/org/openstreetmap/atlas/utilities/testing/TestAtlasHandler.java [hunk 3]
```diff
@@ -143,6 +163,12 @@ else if (fullySplit.size() == 1)
                 throw new CoreException("{} isn't a valid tag description", tagAndValue);
             }
         }
+        // Add a country code if one does not already exist
+        if (!iso.equals(ISOCountryTag.UNKNOWN_ISO_COUNTRY)
```
Comment (nachtm) [inline src/main/java/org/openstreetmap/atlas/utilities/testing/TestAtlasHandler.java]: What do you think about passing around an `Optional<String>` for the ISO code? Then Optional.empty can be the null value which feels a little cleaner than ISOCountryTag.UNKNOWN_ISO_COUNTRY

File: src/test/java/org/openstreetmap/atlas/utilities/testing/TestAtlasTest.java [hunk 4]
```diff
@@ -4,28 +4,56 @@
...
 /**
  * Tests {@link Atlas} creation from both JOSM and Osmosis XML files.
  *
  * @author matthieun
+ * @author bbreithaupt
  */
 public class TestAtlasTest
 {
     private static final Logger logger = LoggerFactory.getLogger(TestAtlasTest.class);
+    private static final String USA = "USA";
```
Comment (nachtm) [inline src/test/java/org/openstreetmap/atlas/utilities/testing/TestAtlasTest.java:21]: 🇺🇸

**Review by nachtm — COMMENTED** [review]
I think this is a really cool addition. Just a couple comments

**Review by danielduhh — COMMENTED** [review]
Looks solid. Approval pending resolution of @nachtm's comments

File: src/main/java/org/openstreetmap/atlas/utilities/testing/TestAtlasHandler.java (hunk 1 above)
Comment (Bentleysb) [reply src/main/java/org/openstreetmap/atlas/utilities/testing/TestAtlasHandler.java]: I don't think it should really have any impact because this is using rectangle.MAXIMUM as the one "country" in the country boundary map. So no actual slicing should occur. But I would agree it is worth calling out.

File: src/main/java/org/openstreetmap/atlas/utilities/testing/TestAtlasHandler.java (hunk 2 above)
Comment (Bentleysb) [reply src/main/java/org/openstreetmap/atlas/utilities/testing/TestAtlasHandler.java:134]: Because of the addition of the iso argument to parseTags() I would have had to pass the iso through to this function. The only thing currently using this function is addArea(), so I thought it was cleaner to have parseTags() called on the result of this function from addArea().

**Review by nachtm — APPROVED** [review]

**Review by sayas01 — APPROVED** [review]
LGTM!

**Comment by Bentleysb** [comment]
@danielduhh ping

**Comment by danielduhh** [comment]
This is good. you got 2 approvals! @Bentleysb
